src/views/Orders.vue

```diff
@@ -217,23 +222,39 @@ export default defineComponent({
         this.getPackedOrders().then(() => { event.target.complete() });
       }
     },
-    async viewOrder (order: any) {
+    async viewOrder (order: any, part: any) {
       // TODO: find a better approach to handle the case that when in open segment we can click on
       // order card to route on the order details page but not in the packed segment
       await this.store.dispatch('order/updateCurrent', { order }).then(() => {
-        this.$router.push({ path: `/orderdetail/${order.orderId}` })
+        this.$router.push({ path: `/orderdetail/${order.orderId}/${part.orderPartSeqId}` })
       })
     },
     async getPickupOrders (vSize?: any, vIndex?: any) {
       const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
       const viewIndex = vIndex ? vIndex : 0;
       const payload = {
-        sortBy: 'orderDate',
-        sortOrder: 'Desc',
-        viewSize,
-        viewIndex,
-        facilityId: this.currentFacility.facilityId
+        "json": {
```

Comment:
I think we should handle it in actions. Also, we should create helper method pass parameters and prepare query.

Reply:
Done Sir